File: DTeacher/app/src/main/java/com/dreamwalker/knu2018/dteacher/BaseApplication.java
package com.dreamwalker.knu2018.dteacher;

import android.app.Application;
import android.content.Context;

 import com.dreamwalker.knu2018.dteacher.UIUtils.DrawUtil;


/**
 * application
 */
public class BaseApplication extends Application {
    protected static BaseApplication sInstance;

    public BaseApplication() {
        sInstance = this;
    }

    @Override
    public void onCreate() {
        super.onCreate();
        DrawUtil.resetDensity(this);
    }


    public static Context getAppContext() {
        return sInstance;
    }
}

File: DTeacher_kt/app/src/main/java/com/dreamwalker/knu2018/dteacher/BaseApplication.kt
package com.dreamwalker.knu2018.dteacher

import android.app.Application
import android.content.Context

import com.dreamwalker.knu2018.dteacher.UIUtils.DrawUtil

/**
 * application
 */
class BaseApplication : Application() {
    init {
        sInstance = this
    }

    override fun onCreate() {
        super.onCreate()
        DrawUtil.resetDensity(this)
    }

    companion object {
        protected var sInstance: BaseApplication


        val appContext: Context
            get() = sInstance
    }
}
